server/src/main/java/com/thoughtworks/go/server/messaging/notifications/PluginNotificationMessageListener.java

```diff
@@ -14,62 +14,44 @@
 import com.thoughtworks.go.plugin.api.response.Result;
+import com.thoughtworks.go.server.messaging.GoMessageListener;
 import com.thoughtworks.go.serverhealth.HealthStateScope;
 import com.thoughtworks.go.serverhealth.HealthStateType;
 import com.thoughtworks.go.serverhealth.ServerHealthService;
 import com.thoughtworks.go.serverhealth.ServerHealthState;
 import org.apache.commons.lang.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
 
-import java.util.Set;
+public class PluginNotificationMessageListener implements GoMessageListener<NotificationMessage> {
+    private NotificationExtension notificationExtension;
+    private ServerHealthService serverHealthService;
+    private static final Logger LOGGER = LoggerFactory.getLogger(PluginNotificationMessageListener.class);
 
-@Component
-public class PluginNotificationService {
-    private static final Logger LOGGER = LoggerFactory.getLogger(PluginNotificationService.class);
-
-    private final NotificationExtension notificationExtension;
-    private final NotificationPluginRegistry notificationPluginRegistry;
-    private final ServerHealthService serverHealthService;
-
-    @Autowired
-    public PluginNotificationService(NotificationExtension notificationExtension, NotificationPluginRegistry notificationPluginRegistry, ServerHealthService serverHealthService) {
+    public PluginNotificationMessageListener(NotificationExtension notificationExtension, ServerHealthService serverHealthService) {
         this.notificationExtension = notificationExtension;
-        this.notificationPluginRegistry = notificationPluginRegistry;
         this.serverHealthService = serverHealthService;
     }
 
-    public <T> void notifyPlugins(PluginNotificationMessage<T> pluginNotificationMessage) throws Exception {
-        Set<String> interestedPlugins = notificationPluginRegistry.getPluginsInterestedIn(pluginNotificationMessage.getRequestName());
-
-        if (interestedPlugins != null && !interestedPlugins.isEmpty()) {
-            for (String interestedPlugin : interestedPlugins) {
-                notifyPlugin(interestedPlugin, pluginNotificationMessage);
-            }
-        }
-    }
-
-    private <T> void notifyPlugin(String pluginId, PluginNotificationMessage<T> pluginNotificationMessage) {
-        HealthStateScope scope = HealthStateScope.forPlugin(pluginId);
+    @Override
+    public void onMessage(NotificationMessage message) {
+        HealthStateScope scope = HealthStateScope.forPlugin(message.pluginId());
         try {
-            Result result = notificationExtension.notify(pluginId, pluginNotificationMessage.getRequestName(), pluginNotificationMessage.getData());
+            Result result = notificationExtension.notify(message.pluginId(), message.getData().getRequestName(), message.getData().getData());
```

Comment:
I think its better for `NotificationMessage` to have `requestName` and actual notification `data`